src/ser.rs

```diff
@@ -479,8 +479,7 @@ where
     }
 }
 
-#[doc(hidden)]
```

Comment:
Keep the `#[doc(hidden)]` ?

Reply:
It was previously redundant, but you're right, it's now relevant.

I was fooled by [this regression in rustc](https://github.com/rust-lang/rust/issues/34537#issuecomment-1511773330) — but I see that as the older rustc in CI finds, this is indeed incorrect.  I'm a bit loath to expose `BcsDeserializer` though, which looks to be quite internal, so I'll look at refactoring it when I get a moment (as I have a suspicion it can be broken down into `Read`/`BufRead` + `serde::Deserializer` — or even if it can't completely we can expose those instead of the internal `BcsDeserializer`).